tldw_Server_API/app/core/DB_Management/ChaChaNotes_DB.py

```
@@ -3166,6 +3248,146 @@ def _map_table_for_backend(self, table_name: str) -> str:
+        to_note_id: str,
+        directed: bool = False,
+        weight: Optional[float] = 1.0,
+        metadata: Optional[Dict[str, Any]] = None,
+        created_by: str,
+    ) -> Dict[str, Any]:
+        """Create a manual note edge, enforcing undirected canonicalization and uniqueness.
+
+        Returns the created edge row as a dict. Raises ConflictError on duplicates.
+        """
+        if not user_id or not from_note_id or not to_note_id or created_by is None or created_by == "":
+            raise InputError("user_id, from_note_id, to_note_id, and created_by are required")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)